include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,686 @@
+   * in this quadrature point. If @p use_upstream_value is true, the shape
+   * functions from the first cell of the interface is used.
+   *
+   * In other words, this function returns the limit of the value of the shape
+   * function in the given quadrature point when approaching it from one of the
+   * two cells of the interface.
+   *
+   * @note This function is typically used to pick the upstream or downstream
+   * value based on a direction. This can be achieved by using
+   * <code>(direction * normal)>0</code> as the first argument of this
+   * function.
+   */
+  double
+  shape_value(const bool         use_upstream_value,
+              const unsigned int interface_dof_index,
+              const unsigned int q_point,
+              const unsigned int component = 0) const;
+
+  /**
+   * Return the jump $[u]=u_{\text{cell1}} - u_{\text{cell2}}$ on the
+   * interface
+   * for the shape function @p interface_dof_index at the quadrature point
+   * @p q_point of component @p component.
+   *
+   * If this is a boundary face (at_boundary() returns true), then
+   * $[u]=u_{\text{cell1}}$.
+   */
+  double
+  jump(const unsigned int interface_dof_index,
+       const unsigned int q_point,
+       const unsigned int component = 0) const;
+
+  /**
+   * Return the average $\{u\}=frac{1}{2}u_{\text{cell1}} +
+   * \frac{1}{2}u_{\text{cell2}}$ on the interface
+   * for the shape function @p interface_dof_index at the quadrature point
+   * @p q_point of component @p component.
+   *
+   * If this is a boundary face (at_boundary() returns true), then
+   * $\{u\}=u_{\text{cell1}}$.
+   */
+  double
+  average(const unsigned int interface_dof_index,
+          const unsigned int q_point,
+          const unsigned int component = 0) const;
+
+private:
+  /**
+   * The list of DoF indices for the current interface, filled in reinit().
+   */
+  std::vector<types::global_dof_index> interface_dof_indices;
+
+  /**
+   * The mapping from interface dof to the two local dof indices of the
+   * FeFaceValues objects. If an interface DoF is only active on one of the
+   * cells, the other one will have numbers::invalid_unsigned_int.
+   */
+  std::vector<std::array<unsigned int, 2>> dofmap;
+
+  /**
+   * The FEFaceValues object for the current cell.
+   */
+  FEFaceValues<dim> internal_fe_face_values;
+
+  /**
+   * The FEFaceValues object for the current cell if the cell is refined.
+   */
+  FESubfaceValues<dim> internal_fe_subface_values;
+
+  /**
+   * The FEFaceValues object for the neighboring cell.
+   */
+  FEFaceValues<dim> internal_fe_face_values_neighbor;
+
+  /**
+   * The FEFaceValues object for the neighboring cell if the cell is refined.
+   */
+  FESubfaceValues<dim> internal_fe_subface_values_neighbor;
+
+  /**
+   * Pointer to internal_fe_face_values or internal_fe_subface_values,
+   * respectively as determined in reinit().
+   */
+  FEFaceValuesBase<dim> *fe_face_values = nullptr;
```

Comment:
Just initialize it in the constructor as we usually do. Same for the next variable.

Reply:
(actually, you already do that, so just drop the initializer here.)